subsys/net/lib/lwm2m_client_utils/lwm2m/location_assistance_obj.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_obj_loc_assistance
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <string.h>
+#include <init.h>
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include <net/lwm2m_client_utils.h>
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS)
+#include <net/nrf_cloud_agps.h>
+#endif
+
+#define LOCATION_ASSIST_VERSION_MAJOR 1
+#define LOCATION_ASSIST_VERSION_MINOR 0
+
+/* Assistance types */
+#define ASSISTANCE_REQUEST_TYPE_IDLE			0
+#define ASSISTANCE_REQUEST_TYPE_SINGLECELL_INFORM	1
+#define ASSISTANCE_REQUEST_TYPE_MULTICELL_INFORM	2
+#define ASSISTANCE_REQUEST_TYPE_SINGLECELL_REQUEST	3
+#define ASSISTANCE_REQUEST_TYPE_MULTICELL_REQUEST	4
+#define ASSISTANCE_REQUEST_TYPE_AGPS			5
+#define ASSISTANCE_REQUEST_TYPE_PGPS			6
+
+/* Location Assistance resource IDs */
+#define LOCATION_ASSIST_ASSIST_TYPE			0
+#define LOCATION_ASSIST_AGPS_MASK			1
+#define LOCATION_ASSIST_PGPS_PRED_COUNT			2
+#define LOCATION_ASSIST_PGPS_PRED_INTERVAL		3
+#define LOCATION_ASSIST_PGPS_START_GPS_DAY		4
+#define LOCATION_ASSIST_PGPS_START_GPS_TIME_OF_DAY	5
+#define LOCATION_ASSIST_ASSIST_DATA			6
+#define LOCATION_ASSIST_RESULT_CODE			7
+#define LOCATION_ASSIST_LATITUDE			8
+#define LOCATION_ASSIST_LONGITUDE			9
+#define LOCATION_ASSIST_ALTITUDE			10
+#define LOCATION_ASSIST_ACCURACY			11
+
+#define LOCATION_ASSIST_MAX_ID				12
+
+static int32_t assist_type;
+static int32_t agps_mask;
+static int32_t pgps_pred_count;
+static int32_t pgps_pred_interval;
+static int32_t pgps_start_gps_day;
+static int32_t pgps_start_gps_time_of_day;
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS)
+static char assist_buf[4096];
```

Comment:
Should these be configurable?  Or at least set through the Kconfig and provide some reasoning for the chosen values even though there wouldn't be a need to change those.